clusterloader2/cmd/clusterloader.go

```diff
@@ -251,25 +256,20 @@ func main() {
 	junitReporter := ginkgoreporters.NewJUnitReporter(path.Join(clusterLoaderConfig.ReportDir, "junit.xml"))
 	junitReporter.SpecSuiteWillBegin(ginkgoconfig.GinkgoConfig, suiteSummary)
 	testsStart := time.Now()
-	for _, clusterLoaderConfig.TestConfigPath = range testConfigPaths {
-		testStart := time.Now()
-		specSummary := &ginkgotypes.SpecSummary{
-			ComponentTexts: []string{suiteSummary.SuiteDescription, clusterLoaderConfig.TestConfigPath},
+	if testSuiteConfigPath != "" {
+		testSuite, err := config.GetTestSuite(testSuiteConfigPath)
+		if err != nil {
+			klog.Exitf("Error while reading test suite: %v", err)
 		}
-		printTestStart(clusterLoaderConfig.TestConfigPath)
-		if errList := test.RunTest(f, prometheusFramework, &clusterLoaderConfig); !errList.IsEmpty() {
-			suiteSummary.NumberOfFailedSpecs++
-			specSummary.State = ginkgotypes.SpecStateFailed
-			specSummary.Failure = ginkgotypes.SpecFailure{
-				Message: errList.String(),
-			}
-			printTestResult(clusterLoaderConfig.TestConfigPath, "Fail", errList.String())
-		} else {
-			specSummary.State = ginkgotypes.SpecStatePassed
-			printTestResult(clusterLoaderConfig.TestConfigPath, "Success", "")
+		for i := range testSuite {
+			clusterLoaderConfig.TestOverridesPath = testSuite[i].OverridePaths
+			clusterLoaderConfig.TestConfigPath = testSuite[i].ConfigPath
+			runSingleTest(f, prometheusFramework, junitReporter, suiteSummary)
+		}
+	} else {
+		for _, clusterLoaderConfig.TestConfigPath = range testConfigPaths {
```

Comment:
this is super hacky  - can you make this assignment explicit

Reply:
Done.